### ISSUE TITLE
Tree shaking 

### ISSUE THREAD

**Issue by sayjeyhi** [styled-components/polished#478]
Hi there,

I was analyzing my packages and figure out that all of polished function did imported and impact my bundle size.
I just used `darken` and `lighten` functions. 
here is bundlephobia which tell all single function has same size as all package :
https://bundlephobia.com/result?p=polished@3.4.4

**Comment by bhough** [comment]
@sayjeyhi polished is fully tree-shakeable and has been for a couple years now.. FWIW I don't think Bundlephobia is evidence that something isn't properly tree-shakeable. Here is another example where it is incorrect with styled-components: https://bundlephobia.com/result?p=styled-components@5.0.1.

 We will need more information about your setup to provide any help on why it isn't working in your particular instance:

- Version of polished you are using.
- Example of how you are using/importing the functions.
- Which bundler and version you are using.
- Config for said bundler.

Our docs also link to how to set this up in webpack and rollup: https://github.com/styled-components/polished#usage

**Comment by sayjeyhi** [comment]
so you say this is just for lighten and darken functions : 
<img width="775" alt="Screen Shot 1398-12-14 at 11 29 37 PM" src="https://user-images.githubusercontent.com/6254009/75917943-1501a980-5e70-11ea-9b7a-e2dd14624397.png">

I am using webpack 4.41.2 , and I did import polished in es module way(`import {foo} from 'bar'`) and my polished version is 3.4.2

**Comment by sayjeyhi** [comment]
<img width="711" alt="Screen Shot 1398-12-15 at 10 14 21 AM" src="https://user-images.githubusercontent.com/6254009/75955259-b9183e80-5eca-11ea-99da-3ed5751d6006.png">

I import `lighten` by adding full path and bundle size gets better result.
```javascript
import lighten from 'polished/lib/color/lighten';
```

**Comment by avin-kavish** [comment]
So, is there a solution for this problem except using sub-module imports? Why does tree shaking break on root level imports only on this library?

**Comment by bhough** [comment]
@avin-kavish This is still an open issue @sayjeyhi is just suggesting an interim solution. We've been unable to recreate this yet. If you'd like to provide the info we asked the OP for, we can continue to dig into it.

**Comment by sayjeyhi** [comment]
@bhough maybe I could work around it this weekend and come with a PR...

**Comment by bhough** [comment]
@sayjeyhi All we really need is a link to a repo we can fork that shows where it isn't tree-shaking and can take it from there.

**Comment by bhough** [comment]
![image](https://user-images.githubusercontent.com/769339/79701464-4ccf7f80-826b-11ea-9bb4-b43255a47ef4.png)

@sayjeyhi  the screenshot is `lighten` and `darken` using a basic CRA app out of the box. The first difference I see is your build is pulling from `lib` where mine is pulling from `dist`, so that is likely a good place for us to start.

**Comment by avin-kavish** [comment]
Yeah, it works with minimal code, trouble happens when there's a complex tree of dependencies.

```
polished
|-grommet (another npm package)
   |-Layout A
   |  |-Page A
   |-Layout B
      |-Page B
```

So I created a tree like this in Next.js (which is where I am having the trouble) but I can't get an accurate size because of module concatenation. I will keep experimenting...

**Comment by bhough** [comment]
> Edit: With module concat turned off, it's only adding 2.5kB so no go once again.

Not sure what you mean by 'no go once again'?  The fact that it is only adding 2.5kB should mean it is working correctly. Keep in mind color modules are relatively heavy given their dependencies.

**Comment by avin-kavish** [comment]
I was trying to create a reproduction of the issue in a code base separate from the one I'm having the issue in. I can't share the original as it is not open source. What I mean is that the issue is likely due to some weird interaction between dependencies etc in much larger/complex code base.

**Comment by bhough** [comment]
K @avin-kavish We will keep this up for the time being to see if we can get a recreation. Only issue we've seen is folks pulling in the non-es-module version from the `main` field and thus not getting tree-shaking. We are going to do a bit of work to make that less impactful when it happens.

**Comment by avin-kavish** [comment]
I see. I don't think that's happened in my case based on the screenshot below. Maybe the following info could be helpful?

### yarn why polished

```
=> Found "polished@3.5.2"
info Reasons this module exists
   - "grommet" depends on it
   - Hoisted from "grommet#polished"
info Disk size without dependencies: "3.49MB"
info Disk size with unique dependencies: "4.17MB"
info Disk size with transitive dependencies: "4.21MB"
info Number of shared dependencies: 2
```

### Bundle Analyzer

Full Bundle with Polished: **432kb.**
Full Bundle without Polished: **422kb.**

![Untitled](https://user-images.githubusercontent.com/48435155/79778407-4ff36a00-8356-11ea-8dab-56d96fcec2c0.jpg)


### How did I remove only `polished` if my project depends on `grommet`, a 3rd party library which depends on `polished`?

I created my own fork of `grommet` and removed all references to `polished`. Luckily, on the `rgba` function was imported 3 times in the whole project. No other functions were imported. Their use case was trivial enough to replace with strings. In one situation, I broke a certain component, but since I don't use that component in my project, it's no issue for me.

**Comment by bhough** [comment]
@avin-kavish That's good context. The fact that it is being pulled in from another deps could be significant. Could be something `grommet` is doing in their implementation that keeps it from being tree-shaken. Will explore this further.

**Comment by avin-kavish** [comment]
Could it be this?

```
exports.__esModule = true;

var _components = require("./components");

Object.keys(_components).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  exports[key] = _components[key];
});
```

Their exports seems to be dynamically defined.

**Comment by twdrake-ap** [comment]
I'm running into this issue as well, `import { darken, lighten } from 'polished';` is causing the whole package to be bundled with a basic webpack config. I'm on version 3.6.6.

**Comment by langri-sha** [comment]
☝️ same issue as @twdrake-ap. On a basic Gatsby project, using `import { transparentize } from 'polished`, I get 100 modules concatenated together with the main package module.

<details>
  <img src="https://user-images.githubusercontent.com/77084/101861003-606b8580-3b6f-11eb-8317-5716d742f414.png" />
</details>

**Comment by Armanio** [comment]
Same issue. 🤔 
If using direct import with full path works fine.

Simple [workaround](https://gist.github.com/Armanio/2b5bfdbf5dfa1b67d6cf5931170ba4f8).
before:
<img width="813" alt="Снимок экрана 2021-02-25 в 18 52 56" src="https://user-images.githubusercontent.com/3195714/109181348-e1184600-779c-11eb-9e07-d96f199fa4c7.png">

after:
<img width="333" alt="Снимок экрана 2021-02-25 в 18 51 34" src="https://user-images.githubusercontent.com/3195714/109181344-dfe71900-779c-11eb-9772-749949af8df6.png">

**Comment by sbdchd** [comment]
@hipstersmoothie disabling [source-maps](https://webpack.js.org/configuration/devtool/#root) is a no go for me because they're needed for debugging errors in production

**Comment by Profesor08** [comment]
`export default` in every module is causing this issue when you import some function with `import { fn } from "polished";`. Avoiding of using default exports will solve this issue.
